course4/homework-string/string.cc

```diff
@@ -1,9 +1,117 @@
 #include "string.hh"
+#include <sstream>
+#include <iostream>
 
 namespace course {
 
 string::string()
 {
+    mLength = 0;
 }
 
-}
\ No newline at end of file
+string::string(const char* chars)
+{
+    mLength = getLenghtOfChars(chars);
+    mCharArray = std::make_unique<char[]>(mLength);
+    deepCopy(chars, mLength);
+}
+
+string::string(const string& source)
+{
+    mLength = source.size();
+    mCharArray = std::make_unique<char[]>(mLength);
+    deepCopy(source.mCharArray.get(), mLength);
+}
+
+char& string::operator[](int index) const
```

Comment:
With this implementation, you are allowing for changing the content of const object. You are breaking the promise of constness.